--- origsrc/tk8.5.6/generic/default.h	2004-03-17 12:15:42.000000000 -0600
+++ src/tk8.5.6/generic/default.h	2009-03-15 02:17:10.062500000 -0500
@@ -16,8 +16,7 @@
 #ifndef _DEFAULT
 #define _DEFAULT
 
-#if defined(__WIN32__) || defined(_WIN32) || \
-    defined(__CYGWIN__) || defined(__MINGW32__)
+#if defined(__WIN32__) || defined(_WIN32) || defined(__MINGW32__)
 #   include "tkWinDefault.h"
 #else
 #   if defined(MAC_OSX_TK)
--- origsrc/tk8.5.6/unix/configure.in	2008-12-22 12:52:02.000000000 -0600
+++ src/tk8.5.6/unix/configure.in	2009-03-15 02:23:01.953125000 -0500
@@ -646,8 +646,8 @@ else
         eval "TK_EXP_FILE=libtk${TCL_EXPORT_FILE_SUFFIX}"
     
         if test "$GCC" = "yes" ; then
-            TK_BUILD_LIB_SPEC="-Wl,-bI:`pwd | sed -e 's/ /\\\\ /g'`/${TK_BUILD_EXP_FILE} -L`pwd | sed -e 's/ /\\\\ /g'`"
-            TK_LIB_SPEC="-Wl,-bI:${libdir}/${TK_EXP_FILE} -L${libdir}"
+            TK_BUILD_LIB_SPEC="-Wl,`pwd | sed -e 's/ /\\\\ /g'`/${TK_BUILD_EXP_FILE} -L`pwd | sed -e 's/ /\\\\ /g'`"
+            TK_LIB_SPEC="-Wl,${libdir}/${TK_EXP_FILE} -L${libdir}"
         else
             TK_BUILD_LIB_SPEC="-bI:`pwd | sed -e 's/ /\\\\ /g'`/${TK_BUILD_EXP_FILE}"
             TK_LIB_SPEC="-bI:${libdir}/${TK_EXP_FILE}"
